### PR TITLE
Support a more recent version of live555, support of setting SDP range clock

### PR DESCRIPTION
**Pull request by s-beard**
- Use a more recent live555 version 17-03-2021
- Add a MediaSessionHelper class to extend the live MediaSession for add possibility to push a range clock to the SDP.
- adapt mkvclient to new live555 types
- RTSPClientConnection :
   * get the starttime uri parameters to push it into the range clock SDP
   * to get stream with range clock use the same process than VLC : START, PAUSE AND Start with range clock informations.

### CODE REVIEW

**Comment by mpromonet** [comment]
Hi s-beard,

Thanks for the PR, I will look at this, it will review the modification.

Best Regards,
Michel.

File: src/rtspconnectionclient.cpp [hunk 1]
```diff
@@ -235,17 +292,61 @@ void RTSPConnection::RTSPClientConnection::continueAfterPLAY(int resultCode, cha
...
 	}
 	envir().taskScheduler().unscheduleDelayedTask(m_ConnectionTimeoutTask);
 	delete[] resultString;
 }
 
+void RTSPConnection::RTSPClientConnection::continueAfterPAUSE(int resultCode, char* resultString)
+{
+	if (resultCode != 0)
+	{
+		envir() << "Failed to PLAUSE: " << resultString << "\n";
```
Comment (mpromonet) [inline src/rtspconnectionclient.cpp:315]: Failed to PAUSE ?

File: src/MediaSessionHelper.cpp [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+You should have received a copy of the GNU Lesser General Public License
+along with this library; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301  USA
+**********/
+// "liveMedia"
+// Copyright (c) 1996-2019 Live Networks, Inc.  All rights reserved.
+// A data structure that represents a session that consists of
+// potentially multiple (audio and/or video) sub-sessions
+// Implementation
```
[the remark this answers is not in the document]
Comment (s-beard) [reply src/MediaSessionHelper.cpp]: i've got removed this file

File: inc/MediaSessionHelper.h [hunk 3]
```diff
@@ -0,0 +1,18 @@
+
+
+#include "MediaSession.hh"
+
+class MediaSessionHelper : public MediaSession
+{
+public:
+	//MediaSessionHelper(UsageEnvironment &env) : MediaSession(UsageEnvironment env);
```
[the remark this answers is not in the document]
Comment (s-beard) [reply inc/MediaSessionHelper.h]: i've got removed this file

File: src/MediaSessionHelper.cpp [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+#include "MediaSessionHelper.h"
+#include "MediaSession.hh"
+
+#include <iostream>
+#include <sstream>
+
+MediaSessionHelper::MediaSessionHelper(UsageEnvironment &env) : MediaSession(env)
+{
+	std::cout << " MediaSessionHelper constructor \n";
```
[the remark this answers is not in the document]
Comment (s-beard) [reply src/MediaSessionHelper.cpp]: i've got removed this file

File: src/MediaSessionHelper.cpp [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+		}
+	}
+
+	return newSession;
+}
+
+// Force to set the start time and the end time at format "YYYMMDDTHHMMSSZ"
+// use when SDP do not contains a=range: clock, but can use it, [ CASD Visimax ]
+void MediaSessionHelper::setRangeClockAttribute(const char *absStartTime, const char *absEndTime)
+{
```
[the remark this answers is not in the document]
Comment (s-beard) [reply src/MediaSessionHelper.cpp]: i've got removed this file

**Pull request merged**